aten/src/ATen/native/cuda/Loops.cuh

```diff
@@ -110,7 +110,7 @@ struct AUnaryFunctor {
   using traits = function_traits<func_t>;
   using opmath_arg1_t = typename traits::template arg<0>::type;
   __device__ return_t operator()(arg2_t b) const {
-    return f(a, b);
+    return static_cast<return_t>(f(a, b));
```

Comment:
We only allow explicit conversion from `complex<float>` to `complex<Half>`

Reply:
what's the reason behind this?

Reply:
Since it is a downcast, I think it would be better to have it explicit.

Reply:
Is that also true for `Half`?

Reply:
`Half` supports implicit casting from `float`
https://github.com/pytorch/pytorch/blob/4b311a9633f721f4d60d9cd096959a8e5774f554/c10/util/Half.h#L386

Reply:
I think we should have consistent rules for both these data types. cc @ngimel what are your thoughts?

Reply:
Be consistent

Reply:
Have removed the explicit from constructor (complex<float> can now be implicitly casted to complex<half>). Have also removed the `static_cast`s from `Loops.cuh`.